pkg/loop/internal/reporting_ocr3.go

```diff
@@ -0,0 +1,393 @@
+		ReportWithInfo: pbReportsWithInfo(ri),
+	}, nil
+}
+
+func (o *ocr3reportingPluginServer) ShouldAcceptAttestedReport(ctx context.Context, request *pb.OCR3ShouldAcceptAttestedReportRequest) (*pb.OCR3ShouldAcceptAttestedReportReply, error) {
+	sa, err := o.impl.ShouldAcceptAttestedReport(ctx, request.SegNr, ocr3types.ReportWithInfo[any]{
+		Report: request.Ri.Report,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &pb.OCR3ShouldAcceptAttestedReportReply{
+		ShouldAccept: sa,
+	}, nil
+}
+
+func (o *ocr3reportingPluginServer) ShouldTransmitAcceptedReport(ctx context.Context, request *pb.OCR3ShouldTransmitAcceptedReportRequest) (*pb.OCR3ShouldTransmitAcceptedReportReply, error) {
+	st, err := o.impl.ShouldTransmitAcceptedReport(ctx, request.SegNr, ocr3types.ReportWithInfo[any]{
+		Report: request.Ri.Report,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &pb.OCR3ShouldTransmitAcceptedReportReply{
+		ShouldTransmit: st,
+	}, nil
+}
+
+func (o *ocr3reportingPluginServer) Close(ctx context.Context, empty *emptypb.Empty) (*emptypb.Empty, error) {
+	return &emptypb.Empty{}, o.impl.Close()
+}
+
+func pbOutcomeContext(oc ocr3types.OutcomeContext) *pb.OCR3OutcomeContext {
+	return &pb.OCR3OutcomeContext{
+		SeqNr:           oc.SeqNr,
+		PreviousOutcome: oc.PreviousOutcome,
+		Epoch:           oc.Epoch, //nolint:all
+		Round:           oc.Round, //nolint:all
+	}
+}
+
+func pbAttributedObservation(ao libocr.AttributedObservation) *pb.OCR3AttributedObservation {
+	return &pb.OCR3AttributedObservation{
+		Observation: ao.Observation,
+		Observer:    uint32(ao.Observer),
+	}
+}
+
+func pbReportsWithInfo(rwi []ocr3types.ReportWithInfo[any]) (ri []*pb.OCR3ReportWithInfo) {
+	for _, r := range rwi {
+		ri = append(ri, &pb.OCR3ReportWithInfo{
+			Report: r.Report,
+		})
+	}
+	return
+}
+
+func pbOcr3AttributedObservations(aos []libocr.AttributedObservation) (pbaos []*pb.OCR3AttributedObservation) {
+	for _, ao := range aos {
+		pbaos = append(pbaos, pbAttributedObservation(ao))
+	}
+
+	return pbaos
+}
+
+func outcomeContext(oc *pb.OCR3OutcomeContext) ocr3types.OutcomeContext {
+	return ocr3types.OutcomeContext{
+		SeqNr:           oc.SeqNr,
+		PreviousOutcome: oc.PreviousOutcome,
+		Epoch:           oc.Epoch, //nolint:all
+		Round:           oc.Round, //nolint:all
+	}
+}
+
+func ocr3AttributedObservation(pbo *pb.OCR3AttributedObservation) (o libocr.AttributedObservation, err error) {
+	o.Observation = pbo.Observation
+	if pbo.Observer > math.MaxUint8 {
+		err = ErrUint8Bounds{Name: "Observer", U: pbo.Observer}
+		return
+	}
+	o.Observer = commontypes.OracleID(pbo.Observer)
+	return
+}
+
+func ocr3AttributedObservations(pbo []*pb.OCR3AttributedObservation) (o []libocr.AttributedObservation, err error) {
+	for _, ao := range pbo {
+		a, err := ocr3AttributedObservation(ao)
+		if err != nil {
+			return nil, err
+		}
+		o = append(o, a)
+	}
+	return
+}
+
+func reportsWithInfo(ri []*pb.OCR3ReportWithInfo) (rwi []ocr3types.ReportWithInfo[any]) {
```

Comment:
@george-dorin I thought you'd discussed passing in a protobuf rather than an any type in the reportWithInfo, is that right?